go-controller/pkg/ovn/ovn.go

```diff
@@ -780,12 +780,29 @@ func (oc *Controller) WatchNetworkPolicy() {
 		AddFunc: func(obj interface{}) {
 			policy := obj.(*kapisnetworking.NetworkPolicy)
 			oc.initRetryPolicy(policy)
+			oc.checkAndSkipRetryPolicy(policy)
+			// If there is a delete entry this is a network policy being added
+			// with the same name as a previous network policy that failed deletion.
+			// Destroy it first before we add the new policy.
+			if oc.retryEntryHasDelete(policy) {
+				klog.Infof("Detected stale policy during new policy add with the same name: %s/%s",
+					policy.Namespace, policy.Name)
+				retryEntry := oc.getRetryEntry(policy)
+				if err := oc.deleteNetworkPolicy(retryEntry.oldPolicy, nil); err != nil {
+					oc.unSkipRetryPolicy(policy)
+					klog.Errorf("Failed to delete stale network policy %s, during add: %v",
+						getPolicyNamespacedName(policy), err)
+					return
+				}
```

Comment:
Shouldn't we clear oldPolicy from the retry entry here on delete success?

Reply:
oh good point.... I guess retryEntry.oldPolicy will still have that old field set huh...yeah we should call removeDeleteFromRetryPolicy here @trozet ?